src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/VariantsSparkSinkUnitTest.java

```diff
@@ -84,6 +94,72 @@ public void testWritingToFileURL(String vcf, String outputFileExtension) throws
         assertSingleShardedWritingWorks(vcf, outputUrl);
     }
 
+    @Test(expectedExceptions = UserException.UnimplementedFeature.class)
+    public void testGVCF_GZ_isDisallowed() throws IOException {
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        VariantsSparkSink.writeVariants(ctx, createTempFile("test",".g.vcf.gz").toString(), null,
+                                        new VCFHeader(), true, Arrays.asList(1, 2, 4, 5), 2, 1 );
+    }
+
+    @DataProvider
+    public Object[][] gvcfCases(){
+        //TODO enable disabled cases when https://github.com/broadinstitute/gatk/issues/4274 is resolved
+        return new Object[][]{
+                {true, ".g.vcf"},
+                {false, ".vcf"},
+            //    {true, "g.vcf.gz"},
+                {false, ".vcf.gz"},
+           //    {true, ".g.bcf"},
+                {false, ".bcf"},
+                {false, ".bcf.gz"},
+            //    {true, ".g.bcf.gz"}
+        };
+    }
+
+    @Test(dataProvider = "gvcfCases")
+    public void testEnableDisableGVCFWriting(boolean writeGvcf, String extension) throws IOException {
+        List<VariantContext> vcs = new ArrayList<>();
+        for(int i = 1; i <= 10; i++) {
+            final Allele A = Allele.create("A", true);
+            final VariantContext vc = new VariantContextBuilder("hand crafted", "1", i, i,
+                                                                Arrays.asList(A, Allele.NON_REF_ALLELE))
+                    .genotypes(new GenotypeBuilder(SAMPLE).alleles(Arrays.asList(A, A)).DP(10).GQ(10).PL(new int[]{0, 60, 10}).make())
+                    .make();
+            vcs.add(vc);
+        }
+
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        final File output = createTempFile(outputFileName, extension);
+        VariantsSparkSink.writeVariants(ctx, output.toString(), ctx.parallelize(vcs), getHeader(), writeGvcf, Arrays.asList(100), 2, 1 );
+
+        new CommandLineProgramTest(){
+            @Override
+            public String getTestedToolName(){
+                return IndexFeatureFile.class.getSimpleName();
+            }
+        }.runCommandLine(new String[]{"-F", output.getAbsolutePath()});
```

Comment:
Isn't `new Main().instanceMain(args)` simpler here?

Reply:
It is but it misses out on some things like setting the default test verbosity.